### PR TITLE
Fix phpcr generics-annotations phpstan issues

### PR DESCRIPTION
**Pull request by alexander-schranz**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no 
| Fixed tickets | fixes # 
| Related issues/PRs | # 
| License | MIT
| Documentation PR | sulu/sulu-docs# 

#### What's in this PR?

This shows a new baseline based on https://github.com/phpcr/phpcr/pull/111

#### Why?

PHPCR defining generics can make our library more robust and show better phpstan issues.

### CODE REVIEW

File: phpstan-baseline.neon [hunk 1]
```diff
@@ -23121,12 +22646,7 @@ parameters:
 			path: src/Sulu/Bundle/PageBundle/Repository/NodeRepository.php
 
 		-
-			message: "#^Method Sulu\\\\Bundle\\\\PageBundle\\\\Repository\\\\NodeRepository\\:\\:getReferences\\(\\) return type has no value type specified in iterable type PHPCR\\\\NodeInterface\\.$#"
-			count: 1
-			path: src/Sulu/Bundle/PageBundle/Repository/NodeRepository.php
-
-		-
-			message: "#^Method Sulu\\\\Bundle\\\\PageBundle\\\\Repository\\\\NodeRepository\\:\\:getReferences\\(\\) should return array\\<PHPCR\\\\NodeInterface\\> but returns array\\.$#"
+			message: "#^Method Sulu\\\\Bundle\\\\PageBundle\\\\Repository\\\\NodeRepository\\:\\:getReferences\\(\\) should return array\\<PHPCR\\\\NodeInterface\\> but returns array\\<string, PHPCR\\\\PropertyInterface\\>\\.$#"
```
Comment (dbu) [inline phpstan-baseline.neon]: this looks a bit worrying, something might be wrong.

getReferences in phpcr returns the list of properties that reference a node.

File: phpstan-baseline.neon [hunk 2]
```diff
@@ -59750,6 +57740,11 @@ parameters:
 			count: 1
 			path: src/Sulu/Component/Webspace/Settings/SettingsManager.php
 
+		-
+			message: "#^Method Sulu\\\\Component\\\\Webspace\\\\Settings\\\\SettingsManager\\:\\:loadStringByWildcard\\(\\) should return array\\<string\\> but returns array\\<string, array\\|string\\>\\.$#"
```
Comment (dbu) [inline phpstan-baseline.neon]: did i get something wrong with anntations? or is this not related to phpcr changes?

**Comment by dbu** [comment]
neat! that is a LOT less errors. i commented on 2 places that i feel look odd and might indicate either an actual bug in sulu or a mistake in the annotations i added.

File: phpstan-baseline.neon (hunk 1 above)
Comment (alexander-schranz) [reply phpstan-baseline.neon]: Okay are `References` not `Nodes` are they really `PropertyInterface`: The code is this one: https://github.com/sulu/sulu/blob/1fc46655a4a14afbb2600b6524ff63153d5fd4e1/src/Sulu/Bundle/PageBundle/Repository/NodeRepository.php#L236C35-L236C57

File: phpstan-baseline.neon (hunk 2 above)
Comment (alexander-schranz) [reply phpstan-baseline.neon]: Not sure how `getString` can be `array|string` here: https://github.com/sulu/sulu/blob/1fc46655a4a14afbb2600b6524ff63153d5fd4e1/src/Sulu/Component/Webspace/Settings/SettingsManager.php#L117

Comment (alexander-schranz) [reply phpstan-baseline.neon]: Looks like an error on our side not handling `multivalues` if I understand correctly: https://github.com/phpcr/phpcr/blob/3307bbd639408f16fd947fc1be1f9cf3410c52a6/src/PHPCR/PropertyInterface.php#L575-L576

Comment (alexander-schranz) [reply phpstan-baseline.neon]: Probably should in phpcr be then `string|string[]`

File: phpstan-baseline.neon (hunk 1 above)
Comment (dbu) [reply phpstan-baseline.neon]: the references themselves are properties. they reference a node which you can get with getNode on the property (or a property that you get with getProperty). if you want the referenced uuid/path, you can use getString on the property.

File: phpstan-baseline.neon (hunk 2 above)
Comment (dbu) [reply phpstan-baseline.neon]: there is a lot of places that should be that. fixed them now.

File: phpstan-baseline.neon (hunk 1 above)
Comment (dbu) [reply phpstan-baseline.neon]: phpdoc of NodeInterface::getReferences:

`This method returns all REFERENCE properties that refer to this node,`

File: src/Sulu/Bundle/PageBundle/Resources/phpcr-migrations/Version202005191116.php [hunk 3]
```diff
@@ -49,6 +56,7 @@ private function upgrade(SessionInterface $session)
         $rows = $queryManager->createQuery($query, 'JCR-SQL2')->execute();
 
         foreach ($rows as $row) {
+            /** @var NodeInterface $node */
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/PageBundle/Resources/phpcr-migrations/Version202005191116.php:59]: Can be null only on `outer join` which we don't use so use phpdoc to overwrite the `NodeInterface|null` in the migration

File: src/Sulu/Bundle/CustomUrlBundle/Resources/phpcr-migrations/Version201904110902.php [hunk 4]
```diff
@@ -42,10 +43,11 @@ private function upgrade(SessionInterface $session)
         $rows = $queryManager->createQuery($query, 'JCR-SQL2')->execute();
 
         foreach ($rows as $row) {
+            /** @var NodeInterface $node */
             $node = $row->getNode();
 
             $property = $node->getProperty('domainParts');
-            $oldDomainParts = \json_decode($property->getValue('domainParts'), true);
+            $oldDomainParts = \json_decode($property->getValue(), true);
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/CustomUrlBundle/Resources/phpcr-migrations/Version201904110902.php:50]: no param exists here.

File: src/Sulu/Bundle/CustomUrlBundle/Resources/phpcr-migrations/Version201904110902.php [hunk 5]
```diff
@@ -54,7 +56,7 @@ private function upgrade(SessionInterface $session)
 
             $newDomainParts = \array_merge($newDomainParts, $oldDomainParts['suffix']);
 
-            $property->remove('domainParts');
+            $property->remove();
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/CustomUrlBundle/Resources/phpcr-migrations/Version201904110902.php:59]: no param exists here.

File: src/Sulu/Component/Webspace/Settings/SettingsManagerInterface.php [hunk 6]
```diff
@@ -46,7 +46,7 @@ public function load($webspaceKey, $key);
      * @param string $webspaceKey
      * @param string $key
      *
-     * @return string
+     * @return string|string[]|null
```
Comment (alexander-schranz) [inline src/Sulu/Component/Webspace/Settings/SettingsManagerInterface.php:49]: multi value fields in phpcr can contain list values so I adopted it as this could happen

**Review by Prokyonn — APPROVED** [review]
